SharkCage/SharedFunctionality/tokenLib/tokenLib.cpp

```diff
@@ -134,24 +146,207 @@ namespace tokenLib {
 			return false;
 		}
 		CloseHandle(userToken);
-		
+
 
 		//add desired group to the token
 		if (!tokenDeconstructed->addGroup(sid)) {
 			wprintf(L"  Cannot add group to a token\n");
 			return false;
 		}
 
-		//generate new access token 
+		//generate new access token
 		if (!tokenDeconstructed->generateToken(token)) {
 			wprintf(L"  Cannot construct a token\n");
 			return false;
 		}
 		return true;
 	}
-}
 
+	DLLEXPORT bool aquireTokenWithPrivilegesForTokenManipulation(HANDLE &token) {
+		auto allProcessesOpt = getAllProcesses();
+		if (!allProcessesOpt.has_value()) {
+			std::wcout << L"Cannot enumerate processes" << std::endl;
+			token = 0;
+			return false;
+		}
+		auto allProcesses = allProcessesOpt.value();
+		auto privilegedProcessesOpt = getProcessesWithBothPrivileges(allProcesses);
+		if (!privilegedProcessesOpt.has_value())
+		{
+			std::wcout << L"Cannot locate process with needed privilege" << std::endl;
+			token = 0;
+			return false;
+		}
+		auto privilegedProcesses = privilegedProcessesOpt.value();
+
+		auto processHandleOpt = getProcessUnderLocalSystem(privilegedProcesses);
+		if (!processHandleOpt.has_value()) {
+			std::wcout << L"Cannot locate process with needed privilege" << std::endl;
+			token = 0;
+			return false;
+		}
+
+		HANDLE processHandle = processHandleOpt.value();
+
+		HANDLE processToken = 0;
```

Comment:
Same here: `nullptr` (there are more instances below, I'm going to stop here :D)